src/sentry/static/sentry/app/components/events/interfaces/threads/threadsSelectorOption.tsx

```diff
@@ -0,0 +1,107 @@
+/* eslint-disable react/prop-types */
+import React from 'react';
+import styled from '@emotion/styled';
+
+import InlineSvg from 'app/components/inlineSvg';
+import Badge from 'app/components/badge';
+import Flex from 'app/components/flex';
+import Text from 'app/components/text';
+import TextOverflow from 'app/components/textOverflow';
+import space from 'app/styles/space';
+import {Event} from 'app/types';
+
+import getThreadDetails, {ThreadDetails, Thread} from './getThreadDetails';
+import ThreadsSelectorOptionLabel, {
+  ThreadsSelectorOptionLabelType,
+} from './threadsSelectorOptionLabel';
+
+const StyledInlineSvgContainer = styled(Flex)(({theme}) => ({
+  color: theme.orange,
+  minHeight: 28,
+}));
+
+const StyledBadge = styled(Badge)({
+  marginRight: space(0.75),
+});
+
+const StyledThreadName = styled(Text)({
+  marginBottom: space(0.5),
+  fontWeight: 600,
+});
+
+const StyledInfo = styled(Flex)({
+  minWidth: 0,
+  marginRight: space(0.5),
+});
+
+const StyledInfoDetails = styled(Flex)({
+  maxWidth: '100%',
+});
+
+const StyledContainer = styled(Flex)({
+  maxWidth: '100%',
+});
+
+const StyledFileName = styled('span')({
+  color: '#2c58a8',
+});
+
+const StyledTextOverflow = styled(TextOverflow)({
+  maxWidth: 235,
+});
+
+export interface ThreadsSelectorOptionProps extends Thread {
+  details?: ThreadDetails;
+}
+
+interface Props {
+  thread: Thread;
+  event: Event;
+  selected?: boolean;
+}
+
+const ThreadsSelectorOption: React.FC<Props> = ({thread, event, selected = false}) => {
+  const threadDetails = getThreadDetails(thread, event);
+  return (
+    <StyledContainer flexGrow={1} alignItems="center">
+      <StyledBadge text={thread.id} />
+      <StyledInfo>
+        {selected ? (
+          <StyledTextOverflow>
+            {threadDetails?.filename ? (
+              <React.Fragment>
+                {`Thread: ${thread.name} - `}
+                <StyledFileName>{threadDetails?.filename}</StyledFileName>
+              </React.Fragment>
+            ) : (
+              thread.name
+            )}
+          </StyledTextOverflow>
+        ) : (
+          <StyledInfoDetails flexDirection="column">
+            <StyledThreadName>{thread.name}</StyledThreadName>
+            {threadDetails &&
+              Object.entries(threadDetails).map(([key, value]) => (
+                <ThreadsSelectorOptionLabel
+                  key={key}
+                  label={key as ThreadsSelectorOptionLabelType}
+                  value={value}
+                />
+              ))}
+          </StyledInfoDetails>
+        )}
+      </StyledInfo>
+      {thread.crashed && (
+        <StyledInlineSvgContainer
+          alignItems="center"
```

Comment:
As evan commented in #16636 - we have been trying to get away from this. One of the reasons is that we now have these CSS/flex props mixed with our components props, which can make it a bit noisy. Since it's already a styled component, we can define the styling in the styled component definition.